static/app/components/profiling/profilingOnboardingSidebar.tsx

```diff
@@ -49,7 +51,37 @@ const PROFILING_ONBOARDING_STEPS = [
   ProductSolution.PROFILING,
 ];
 
-export function ProfilingOnboardingSidebar(props: CommonSidebarProps) {
+export function useProfilingOnboardingDrawer() {
+  const organization = useOrganization();
+  const currentPanel = useLegacyStore(SidebarPanelStore);
+  const isActive = currentPanel === SidebarPanelKey.PROFILING_ONBOARDING;
+  const hasProjectAccess = organization.access.includes('project:read');
+
+  const {openDrawer} = useDrawer();
+
+  useEffect(() => {
+    if (isActive && hasProjectAccess) {
+      openDrawer(() => <DrawerContent />, {
+        ariaLabel: t('Profile Code'),
+        // Prevent the drawer from closing when the query params change
+        shouldCloseOnLocationChange: location =>
+          location.pathname !== window.location.pathname,
+      });
+    }
+  }, [isActive, hasProjectAccess, openDrawer]);
+}
+
+function DrawerContent() {
+  useEffect(() => {
+    return () => {
+      SidebarPanelStore.hidePanel();
+    };
+  }, []);
+
+  return <SidebarContent />;
+}
+
+export function LegacyProfilingOnboardingSidebar(props: CommonSidebarProps) {
```

Comment:
Do you think it's worth adding a @deprecated flag here until we remove this?

Reply:
It's only used in one place but I'll slap it on there